### PR TITLE
IIIF tidy up

### LINKED ISSUE #8904: Remove/tidy code relating to iiif v2
Once #8608 is closed we need to remove/tidy up any code left over from dealing with v2 iiif manifests.

- [ ] the fetchManifest function needs to be updated so it no longer fetches v2 manifests
- [ ] functions remaining in utils/iiif/v2 need to be removed, if no longer required, or moved to utils/iiif/v3
- [ ] the types in services/iiif/types/manifest/v2 and images/v2 should be retired
- [ ] there may be some tidy up work to be done in the transformer, e.g. we may be able to do without isTotallyRestricted and needsModal or deal with them in better ways. Similarly we may be able to refactor getPdf and hasPdfDownload.

### PR DESCRIPTION
**Pull request by davidpmccormick**
☝️ 

Closes #8904

### CODE REVIEW

**Review by alexwlchan — COMMENTED** [review]
Mostly looks good, but couple of areas where it'd be helpful to understand the motivation behind the changes.

File: catalogue/webapp/components/ExpandedImage/ExpandedImage.tsx [hunk 1]
```diff
@@ -150,13 +150,11 @@ const ExpandedImage: FunctionComponent<Props> = ({
...
-          manifestV2: undefined,
-          manifestV3: undefined,
-        }
-      );
-      const { firstCollectionManifestLocation, canvases } = transformedManifest;
+      const transformedManifest =
+        iiifManifest && transformManifest(iiifManifest);
+      const { firstCollectionManifestLocation, canvases } = {
+        ...transformedManifest,
+      };
```
[the remark this answers is not in the document]
Comment (davidpmccormick) [reply catalogue/webapp/components/ExpandedImage/ExpandedImage.tsx:157]: it's possibly `undefined`, so this ensures there's always *some* sort of object to destructure.

File: catalogue/webapp/pages/download.tsx [hunk 2]
```diff
@@ -62,7 +63,10 @@ const DownloadPage: NextPage<Props> = ({
       })
     : [];
 
-  const allDownloadOptions = [...iiifImageDownloadOptions, ...downloadOptions];
+  const allDownloadOptions = downloadOptions && [
+    ...iiifImageDownloadOptions,
+    ...downloadOptions,
+  ];
```
[the remark this answers is not in the document]
Comment (davidpmccormick) [reply catalogue/webapp/pages/download.tsx:69]: this, and the below, were required TypeScript hoops to jump through to keep it happy. I felt like it made sense at the point I committed it, but I can't give you a definitive answer to your question off the top of my head now. I'll take another look

File: catalogue/webapp/pages/download.tsx [hunk 3]
```diff
@@ -104,7 +108,7 @@ const DownloadPage: NextPage<Props> = ({
               )}
             </SpacingComponent>
           )}
-          {license && (
+          {license && title && (
```
[the remark this answers is not in the document]
Comment (davidpmccormick) [reply catalogue/webapp/pages/download.tsx:111]: `title` is possibly undefined, but you're right, it's not a requirement to be defined at this level (until the `getCredit` call below). I'll move it down.

File: catalogue/webapp/pages/download.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (davidpmccormick) [reply catalogue/webapp/pages/download.tsx:69]: on further inspection, you're right and I'm being over-cautious to the point of potentially hiding content here. Will update.

File: catalogue/webapp/pages/download.tsx [hunk 4]
```diff
@@ -91,7 +95,7 @@ const DownloadPage: NextPage<Props> = ({
               {displayTitle}
             </Space>
           </SpacingComponent>
-          {workId && (
+          {workId && allDownloadOptions && (
```
[the remark this answers is not in the document]
Comment (davidpmccormick) [reply catalogue/webapp/pages/download.tsx:98]: Same as above.

File: catalogue/webapp/pages/item.tsx [hunk 5]
```diff
@@ -383,14 +383,9 @@ export const getServerSideProps: GetServerSideProps<Props | AppErrorProps> =
...
-    const transformedManifest = transformManifest(
-      iiifManifest || {
-        manifestV2: undefined,
-        manifestV3: undefined,
-      }
-    );
+    const transformedManifest = iiifManifest && transformManifest(iiifManifest);
 
-    const { isCollectionManifest, manifests } = transformedManifest;
+    const { isCollectionManifest, manifests } = { ...transformedManifest };
```
Comment (alexwlchan) [inline catalogue/webapp/pages/item.tsx:388]: Ditto previous comments, can we skip spreading here?

**Pull request merged**